cmd/createorupdate/update.go

```diff
@@ -43,6 +106,49 @@ func update(ctx context.Context, cs *api.OpenShiftManagedCluster, p api.Plugin)
 	return nil
 }
 
+type scaleOp struct {
```

Comment:
I don't really like this structure at all

Reply:
I don't think it adds anything - if you have it, you need additional unit tests, but if you don't have it, the unit tests aren't really lacking.
Clearly, you need a unit test around the function which detects if a diff between cs and oldCs is scale only.
Otherwise, if you robustly drain all excess VMs before deploy() and health check all new VMs post deploy(), then there is no diff-related code to unit test.  It's more robust and you're just in the desired-state->actual state reconciliation world.

Reply:
>  Clearly, you need a unit test around the function which detects if a diff between cs and oldCs is scale only.

If you have a function that detects you have a scale operation, you still need some extra logic to figure out the direction you are scaling. The point of `isScaleOp` is exactly that, to detect whether the current update is a scale operation and furthermore whether it's up or down. In order to make it unit-testable it has to be its own function. When `isScaleOp` returns, `update` needs to know what scale operation it is about to perform on which scale set, hence `isScaleOp` returns a `map[api.AgentPoolProfileRole]scaleOp` which should always be of size 1 but I don't do any sort of validation about it atm. So I am not sure how to better model this and make as much logic unit-testable as possible, other than use `scaleOp`. Probably holding `count` vs `diff` is more robust but that does not diminish our need for `scaleOp`, does it?